### PR TITLE
Sbhat speedup topology

### PR DESCRIPTION
**Pull request by shreyb**
This is to speed up the parsing of records for the OIM Topology piece.  Creating the dictionary takes about 40 seconds now, and then all lookups are from the dictionary.

### CODE REVIEW

File: src/graccreq/oim/OIMTopology.py
Comment (djw8605) [inline src/graccreq/oim/OIMTopology.py]: This should be compiled once, probably in the `__init__` function.  Then used below.  Having it hear compiles it each time this function is called.

File: src/graccreq/oim/OIMTopology.py
Comment (djw8605) [inline src/graccreq/oim/OIMTopology.py:73]: Why do you not pass the resourcename_elt to the `get_resource_information` function?  That would cut down on the root searches, right?

**Review by djw8605 — CHANGES_REQUESTED** [review]

File: src/graccreq/oim/OIMTopology.py
Comment (shreyb) [reply src/graccreq/oim/OIMTopology.py:73]: I tried that initially, but then the XPath searches returned None when I tried to go to a parent element.  I'm still not sure why that was the case.  It should be possible.

That being said, I figured it was okay for us to grab the resource group-specific info from the resource group element (since we'd have to navigate to that level anyway), then descend into the resource element, and grab the information specific to that.

File: src/graccreq/oim/OIMTopology.py
Comment (shreyb) [reply src/graccreq/oim/OIMTopology.py]: Agreed.  I'll move the compile statement up to the **init** function.

**Comment by djw8605** [comment]
I saw that the tests are now taking 155 seconds.  that seems longer.

**Comment by shreyb** [comment]
Yeah - I would have expected that it takes about 20 or 30 seconds longer - not 70 - given that it does all of the XML parsing ahead of time.

What do you propose as the next step?  Most of the overhead is in storing the data in the dictionary.

**Comment by djw8605** [comment]
If you're happy with what's in the pull request now, then I will gladly merge, build, deploy, and test.

**Comment by shreyb** [comment]
It's definitely faster than before.  Do we want to test on a Friday afternoon, or wait until Monday to merge?

**Comment by djw8605** [comment]
Test now.  The processing is getting backed up as we speak.

**Review by djw8605 — APPROVED** [review]

**Comment by shreyb** [comment]
Sounds good.

**Pull request merged**
